geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -268,6 +278,225 @@ impl IntersectionMatrix {
+            );
+            if a_lessthan_b {
+                // [T*T******]
+                return self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
+                    && self.0[CoordPos::Inside][CoordPos::Outside] != Dimensions::Empty;
+            }
+            if a_greaterthan_b {
+                // [T*****T**]
+                return self.0[CoordPos::Inside][CoordPos::Inside] != Dimensions::Empty
+                    && self.0[CoordPos::Outside][CoordPos::Inside] != Dimensions::Empty;
+            }
+        } else {
+            // dimensions are equal: this is only permitted for line / line
+            if dims_a == Dimensions::OneDimensional && dims_b == Dimensions::OneDimensional {
+                // line / line: [0********]
+                return self.0[CoordPos::Inside][CoordPos::Inside] == Dimensions::ZeroDimensional;
+            } else {
+                return false;
+            }
+        }
+        true
+    }
+
+    /// Returns `true` if geometry A and B "spatially overlap". Two geometries overlap if they have the
+    /// same dimension, their interiors intersect in that dimension. and each has at least one point
+    /// inside the other (or equivalently, neither one covers the other)
+    ///
+    /// ```
+    /// use geo_types::{Polygon, polygon};
+    /// use geo::relate::Relate;
+    ///
+    /// let poly1 = polygon![
+    ///     (x: 125., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 160., y: 160.),
+    ///     (x: 125., y: 179.),
+    /// ];
+    /// let poly2 = polygon![
+    ///     (x: 126., y: 179.),
+    ///     (x: 110., y: 150.),
+    ///     (x: 161., y: 160.),
+    ///     (x: 126., y: 179.),
+    /// ];
+    ///
+    /// let intersection = poly1.relate(&poly2);
+    /// assert_eq!(intersection.is_overlaps(), true);
+    /// ```
+    /// # Notes
+    /// - This predicate is **symmetric**
+    #[allow(clippy::nonminimal_bool)]
+    pub fn is_overlaps(&self) -> bool {
+        // ensure that both geometry dimensions are equal
+        let dims_a = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::Inside][CoordPos::OnBoundary])
+            .max(self.0[CoordPos::Inside][CoordPos::Outside]);
+
+        let dims_b = self.0[CoordPos::Inside][CoordPos::Inside]
+            .max(self.0[CoordPos::OnBoundary][CoordPos::Inside])
+            .max(self.0[CoordPos::Outside][CoordPos::Inside]);
+        if dims_a == dims_b {
+            if dims_a == Dimensions::OneDimensional && dims_b == Dimensions::OneDimensional {
+                // line / line
+                self.0[CoordPos::Inside][CoordPos::Outside] == Dimensions::OneDimensional
+                    && self.0[CoordPos::Inside][CoordPos::Outside] != Dimensions::Empty
+                    && self.0[CoordPos::Outside][CoordPos::Inside] != Dimensions::Empty
+            } else {
+                // points or polygons
+                let ii = self.0[CoordPos::Inside][CoordPos::Inside];
```

Comment:
Similarly, is there still value in these checks now that we have dims_a/dims_b? It's ok to leave them if you think the debug_assert is checking something useful, but I wanted to double check since we no longer use ii/ie/ei to compute results.